crowbar_framework/app/models/api/upgrade.rb

```diff
@@ -991,14 +991,30 @@ def run_online_migrations
 
         save_node_action("Executing nova online migrations on #{node.name}...")
         node.wait_for_script_to_finish(
-          "/usr/sbin/crowbar-run-nova-online-migrations.sh",
-          timeouts[:nova_online_migrations]
+          "/usr/sbin/crowbar-nova-migrations-after-upgrade.sh",
+          timeouts[:online_migrations]
         )
         Rails.logger.info("Nova online migrations finished.")
-        save_node_action("Nova online migrations finished.")
+
+        heat_servers = ::Node.find("roles:heat-server")
+        return if heat_servers.empty?
```

Comment:
is it possible that we have no `roles:nova-controller` but nodes with `roles:heat-server` are present? if so, then heat code will never run because of return above.

Reply:
Is it possible to have heat without nova?

Reply:
Probably is (although unlikely). PR adapted.